pkg/library/home/persistentHome.go

```diff
@@ -95,3 +117,107 @@ func storageStrategySupportsPersistentHome(workspace *common.DevWorkspaceWithCon
 	storageClass := workspace.Spec.Template.Attributes.GetString(constants.DevWorkspaceStorageTypeAttribute, nil)
 	return storageClass != constants.EphemeralStorageClassType
 }
+
+func addInitContainer(dwTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) {
+
+	initContainer := inferInitContainer(dwTemplateSpec)
+	if initContainer == nil {
+		// if cannot infer initcontainer, fail quietly
+		return
+	}
+
+	if !initComponentExists(dwTemplateSpec) {
```

Comment:
The checks are set like this to still give some kind of freedom for the user for the initcontainer. For example, if the devfile author sets a the `init-persistent-home` container component like: https://github.com/devfile/devworkspace-operator/blob/57b69c4baca6017f5a44b9e1f9eea926010f0821/pkg/library/home/testdata/persistent-home/creates-home-vm-initcontainer-already-defined.yaml#L19-L23

and not provide the `init-persistent-home` command or event, the DWO would just add the necessary command and event to make it an initcontainer for the workspace pod.

WDYT?

Reply:
If the devfile already has a container component with the name `init-persistent-home`, what should we do?

Two options are, we can either disable persistent home completely, or we can just keep it and have the DWO create the devfile command (not to be confused with a container's `command` field) and devfile event and still let the persistent home feature work. The code right now just does the latter.

> This seems like it still may allow for users (who have edit rights to the DWOC) to set a custom initContainer and command that would be added to all devworkspaces

Devfile authors can create a devfile with any container image anyways, and I don't see how this is related to DWOC anymore

Reply:
Sorry, to clarify a bit more, users won't be able to

Reply:
> This seems like it still may allow for users (who have edit rights to the DWOC) to set a custom initContainer and command that would be added to all devworkspaces

Sorry, to clarify a bit more, users won't be able to do that because of this update to this PR: https://github.com/devfile/devworkspace-operator/pull/1251/commits/76db7ab5e2d5bc0eed4bc5fe05cb74cea1f84e1e

I think https://github.com/devfile/devworkspace-operator/pull/1251#discussion_r1581323016 is simpler and clearer to the user, and I don't feel too strongly on the current approach, I will update my PR

Reply:
Updated: https://github.com/devfile/devworkspace-operator/pull/1251/commits/3e55cc347d5dacfd0c38f6c25b3da6fdc91acb95